endpoint/endpoint.go

```diff
@@ -38,12 +39,78 @@ func (ttl TTL) IsConfigured() bool {
 	return ttl > 0
 }
 
+// Targets is a representation of a list of targets for an endpoint.
+type Targets []string
+
+// NewTargets is a convenience method to create a new Targets object from a vararg of strings
+func NewTargets(target ...string) Targets {
+	t := make(Targets, 0, len(target))
+	t = append(t, target...)
+	return t
+}
+
+func (t Targets) String() string {
+	return strings.Join(t, ";")
+}
+
+func (t Targets) Len() int {
+	return len(t)
+}
+
+func (t Targets) Less(i, j int) bool {
+	return t[i] < t[j]
+}
+
+func (t Targets) Swap(i, j int) {
+	t[i], t[j] = t[j], t[i]
+}
+
+// Same compares to Targets and returns true if they are completely identical
+func (t Targets) Same(o Targets) bool {
+	if len(t) != len(o) {
+		return false
+	}
+	sort.Stable(t)
+	sort.Stable(o)
+
+	for i, e := range t {
+		if e != o[i] {
+			return false
+		}
+	}
+	return true
+}
+
+// IsLess should fulfill the requirement to compare two targets and chosse the 'lesser' one.
+// In the past target was a simple string so simple string comparison could be used. Now we define 'less'
+// as either being the shorter list of targets or where the first entry is less.
+// FIXME We really need to define under which circumstances a list Targets is considered 'less'
+// than another.
+func (t Targets) IsLess(o Targets) bool {
```

Comment:
👍

Reply:
Name feels a bit awkward, but I am fine with it for this PR if noone has a better name (I don't have right now)

Reply:
I'm fine with the name. I'm wondering about the fixme: It says we should define what less means but at the same time describes it above. Is that FIXME still valid?

Reply:
I think so. At least in the sense that current description is a description of what I implemented and nothing which was discussed. This related pretty directly to my comments on the ConflictResolver.
In the past `less` was something related to string comparison and was clearly defined by GoLang. Now we have something more complex and need to define when a list of targets is `less` than the other. In my opinion this is not as clear as string comparison.
It may be that the current definition is completely fine, then I will remove the FIXME, but I am very open to ideas for the definition if `less`.